File: Java/oop/a2constructors/Student.java
package oop.a2constructors;

public class Student {
    
    String name;
    int age;
    double gpa;

    public Student(String name, int age, double gpa){
        this.name = name;
        this.age = age;
        this.gpa = gpa;
    }

    public void Study(){
        System.out.println(this.name + " Studying");
    }

    public void StudentAge(){
        System.out.println(this.age + " year old");
    }

    public void StudentGpa(){
        System.out.println(this.gpa + "\n");
    }
}

File: Cpp/07-oop/02-Constructors.cpp/Student.cpp
#include "Student.h"
#include <iostream>

Student::Student(std::string name, int age) {
    this->name = name;
    this->age = age;
}

void Student::introduce() {
    std::cout << "My name is " << name << "\n";
    std::cout << "I am " << age << " years old.\n";
}

// Student::introduce()
// introduce() เป็นของ Class student

// this->name   → ตัวแปร name ของ Object
// name         → parameter ที่ส่งเข้ามา